### ISSUE TITLE
Feature Request: Borderless Windowed Mode

### ISSUE THREAD

**Issue by Zitchas** [endless-sky/endless-sky#6315]
### Problem Description

When trying to screenshot or screen record on Windows, it is extremely problematic to do so when in fullscreen mode. I can rarely get it to actually show what I'm doing, and takes a lot of recording-software-specific trickery to get it to work, if at all.

### Related Issue Links

-

### Desired Solution

Have an additional option, selectable from the settings panel, to switch Endless Sky to "Borderless Windowed Mode."

Borderless Window Mode is commonly found in many games, and is basically the game running in a window, except a window that has no visible elements and fills the entire screen. This has the benefit of appearing to be simply "full screen," but it enables better compatibility with screen shot, recording, and streaming tools. 

This would be beneficial in making Endless Sky friendlier for streamers to play, which could be good for our public image. It would also make it *much* easier for content creators to make footage demonstrating their content (particularly useful for new mechanics and graphical effects).

### Alternative Approaches

-

**Comment by thomasballinger** [comment]
SDL2 seems to provide lots of options for this, although I couldn't find something that both wasn't fullscreen and hid the top menubar on my mac.

If someone has a WIndows machine and can try [this](https://stackoverflow.com/questions/27116152/sdl2-toggle-sdl-window-resizable-state-for-fake-fullscreen) (the commented-out line `SDL_SetWindowResizable(mainWindow, SDL_FALSE);` works in modern SDL2 versions) by replacing `SDL_SetWindowFullscreen(mainWindow, SDL_WINDOW_FULLSCREEN_DESKTOP);` with it in
https://github.com/endless-sky/endless-sky/blob/a0b4ef43f72a5c3b8e13766b28198074250d3524/source/GameWindow.cpp#L394-L401

that could work, but if it does it may need to be Linux and Windows only.

**Comment by thomasballinger** [comment]
Naive fake fullscreen as described above is too tall on mac because it sets the window to the full height of the monitor.

![image](https://user-images.githubusercontent.com/458879/140654982-e3f34800-cbc8-4724-8aed-cef54d8ba96c.png)

**Comment by thomasballinger** [comment]
@Zitchas do you have an idea of how this should be represented in preferences, maybe a checkbox for "use fake fullscreen?"

**Comment by Zitchas** [comment]
A checkbox would work. 

A bit more complex would be adding it into the rotation triggered by F11 so instead of just swapping between window and full screen, have it rotate through windowed -> full screen -> fake full screen.

**Comment by thomasballinger** [comment]
@Zitchas do you prefer one or the other?

**Comment by Zitchas** [comment]
Not really, no. It's not something I change a lot, so I think it really depends on what has the best compatibility across various operating systems.

**Comment by Terin** [comment]
Not sure that the version of SDL currently used by Endless Sky works with that.
![image](https://user-images.githubusercontent.com/4471575/140656157-4a52fe9d-4278-4427-9141-d52c04d0d19b.png)

Although in the case of Mac, my understanding is that it doesn't really need borderless fullscreen, since programs there don't fully take over the screen the way Windows or Linux could.

**Comment by thomasballinger** [comment]
Ah whoops, the local mac build script must grab a newer version than is used in the other builds.

It looks like the dev64 binaries (did @MCOfficer make these?) at https://endless-sky.github.io/win64-dev.zip contain 2.0.3, but this function was added in 2.0.5.

**Comment by MCOfficer** [comment]
>(did @MCOfficer make these?)

MZ did: https://github.com/endless-sky/endless-sky.github.io/commits/master/win64-dev.zip

**Comment by thomasballinger** [comment]
For mac I see `SDL_GetDisplayUsableBounds()` if we don't want to disable this mode for mac.

**Comment by Terin** [comment]
> It looks like the Windows dev64 binaries (did MCOfficer make these?) at https://endless-sky.github.io/win64-dev.zip contain SDL 2.0.3, but this function was added in 2.0.5.

Might be around time to update SDL, then. Using higher DPI on Windows causes Endless Sky's window to blow up and get blurrier, and I think a later version of SDL is also required to override that (through SDL).

**Comment by snoggles** [comment]
As a workaround, I'm running windowed and using Simple Runtime Window Editor (SWRE) > Fake Fullscreen. It's a manual step every launch, but better than the black screen flashes when alt+tabbing from fullscreen on windows 11.

**Comment by Crystal-RainSlide** [comment]
More workaround... I'm on Windows 10 22H2, I found **Windows 10 fullscreen optimization reduce most of the black screen time when <kbd>Alt</kbd> + <kbd>Tab</kbd> from ES's exclusive fullscreen, but this optimization only works when VSync ON**.

For borderless-ify ES on Windows, I found https://github.com/Codeusa/Borderless-Gaming is not working for Endless Sky. When ES is fullscreen, it does nothing. When ES is windowed, it somehow turn it into exclusive fullscreen, instead of borderless windowed fullscreen. Samething happened to Fake fullscreen in SRWE, but SRWE's Remove borders + Taskbar mode works for me.

[Lossless Scaling](https://losslessscaling.com/) works for ES windowed and can be configured to auto. But I'd say it's the wrong tool. Just use [SRWE](https://github.com/dtgDTGdtg/SRWE) mannally when you really need it, like, writing the wiki with only 1 screen.

A final borderless-ify trick is have an always-on overlay, like custom crosshair software or PowerToys mouse crosshair. Would always work.
